docs/decisions/0007-access-control-of-events.rst

```diff
@@ -0,0 +1,51 @@
+Access control of events
+========================
+
+Status
+------
+
+Pending
+
+Context
+-------
+
+`event-routing-backends` can transform filtered edX events in xAPI and Caliper format which can then be routed to configured URLs. Procedure for configuring URLs of learning record consumers can be found in the `documentation`_. A method is required to filter edX events based on event name, course id, organisation id etc. or on any combination of these properties.
+
+Decision
```

Comment:
Looks good to me overall.
A few things missing, which would be good to add:

- Where can one find documentation of all the possible edx events supported and their fields?  If it's not ready yet, you can put a placeholder for a link to it.
- Clarification on the `org_id` field - the fact that this is associated with enterprise users specifically.

Reply:
Link to edX event fields have been added. Regarding `org_id`, @ziafazal  pointed out that `org_id` is actually pulled from course id and has no relationship with enterprise as seen [here](https://github.com/edx/edx-platform/blob/4f4be6538ae008a6d65cd7cc49d776aebfa9e40a/common/djangoapps/track/contexts.py#L93). Nevertheless, the method will provide flexibility to filter directly on `org_id` or content provider id embedded in `course_id`.

Reply:
Thanks for double-checking on where `org_id` comes from.  As you can see, this is highly error-prone since we have multiple types of organizations in the Open edX domain.  To satisfy the enterprise filtering needs, we'll need to add a separate `user_org_id` field in the event, since enterprises can receive events only for their own users.

Reply:
As discussed, enterprise customers will be able to filter their events based on `enterprise_uuid` key in the edX events. Content organisations will either use `org_id` or their identification embedded in `course_id`. Examples of both cases have been added.

Reply:
Sounds good.